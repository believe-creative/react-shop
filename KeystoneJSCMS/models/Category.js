var keystone = require('keystone');
var Types = keystone.Field.Types;

/**
 * Category Model
 * ==================
 */

var Category = new keystone.List('Category', {
	map: { name: 'title' },	
	autokey:{path:'slug', from:'title', unique:true},
});

Category.add({
	title: { type: String, required: true },
	state:{ type: Types.Select, options:'draft,published,archived', default:'draft', index:true},
	department: {type: Types.Relationship, ref:'Department', index:true},
	publishedDate: { type: Types.Date, index: true, dependsOn: { state: 'published' } },
});

Category.relationship({ ref: 'Product', path: 'products', refPath: 'categories' });

Category.register();
