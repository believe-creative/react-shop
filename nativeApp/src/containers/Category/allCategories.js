import React, {Component} from 'react';
import {Platform, StyleSheet, Text, View,Button,ScrollView} from 'react-native';
import { connect } from "react-redux";
import * as Actions from "../../actions";
import { getCookie } from "../../services/helpers";
import NavBar from '../../components/Navbar/navbar';
import { createStackNavigator, createAppContainer } from 'react-navigation';
import Immutable from "immutable";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";
import createReducers from "../../reducers";
import rootSaga from "../../sagas";
import {store} from "../../store";
import ProductList from '../../components/Product/productlist';
import Footer from '../../components/Footer/footer';
import {styles} from '../Home/home-styles';
import AnimatedLoader from "react-native-animated-loader";

class AllCategories extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activePage: 1,
      visible: true
    };
  }
  componentDidMount() {
    if (this.props.categories) {
      Object.values(this.props.categories).map((category, index) => {
        this.props.loadCategoryProducts({
          token: this.props.token,
          departmentId: category.department_id,
          descriptionLength: 120,
          inStartItem: 0,
          inProductsPerPage: 10000
        });
        return category;
      });
    }
  }
  componentWillReceiveProps(){
    if(this.props.categoryProducts){
      this.setState({
        visible: false
      });
    }
  }
  handlePageChange(pageNumber) {
    this.setState({ activePage: pageNumber });
  }
  render() {
    const { visible } = this.state;
    let productsList = [];
    let productCategoriesList = this.props.categoryProducts
      ? Object.values(this.props.categoryProducts)
      : [];
    productCategoriesList.map((category, index) => {
      category.map((item, ind) => {
        productsList.push(item);
        return item;
      });
      return category;
    });
    let totalItemsCount = productsList.length;
    productsList = productsList.sort(function(a, b) {
      if (a.createdAt < b.createdAt) {
        return -1;
      } else if (a.createdAt > b.createdAt) {
        return 1;
      }
      return 0;
    });
    productsList = productsList.splice((this.state.activePage - 1) * 10, 10);
    return (
      <View>
        <NavBar />
      <ScrollView style={styles.body}>
     <Text style={{...styles.h2, ...styles.black, ...styles.cart_page_wraper}}>All Products</Text>
      <View>
      <AnimatedLoader
        visible={visible}
        overlayColor="rgba(255,255,255,0.75)"
        source={require("../../lottie-loader.json")}
        animationStyle={{width: 100, height: 100}}
        speed={2}
      />
      {this.props.categoryProducts ? Object.values(this.props.categoryProducts).map((productsList,index)=>{
          return  (<ProductList products={productsList} key={index} />);
      }) : <Text/>}

      </View>
        <Footer />
      </ScrollView>
      </View>
    );
  }
}

const mapStateToProps = state => {
  return {
    subCategories: state.get("products").subCategories,
    categories: state.get("products").categories,
    categoryProducts: state.get("products").categoryProducts,
    searchitem: state.get("products").searchItem,
    token: state.get("user").token,
    cart: state.get("products").cart
  };
};

const mapStateToDispatch = dispatch => ({
  loadCategoryProducts: data =>
    dispatch(Actions.getCategoryProducts.request(data))
});

export default connect(
  mapStateToProps,
  mapStateToDispatch
)(AllCategories);
