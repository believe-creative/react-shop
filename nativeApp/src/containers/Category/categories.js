/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, {Component} from 'react';
import {Platform, StyleSheet, Text, View,Button,ScrollView} from 'react-native';
import { connect } from "react-redux";
import * as Actions from "../../actions";
import { getCookie } from "../../services/helpers";
import NavBar from '../../components/Navbar/navbar';
import { createStackNavigator, createAppContainer } from 'react-navigation';
import Immutable from "immutable";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";
import createReducers from "../../reducers";
import rootSaga from "../../sagas";
import {store} from "../../store";
import ProductList from '../../components/Product/productlist';
import Footer from '../../components/Footer/footer';
import {styles} from '../Home/home-styles';
import AnimatedLoader from "react-native-animated-loader";
type Props = {};

class Categories extends Component {
  constructor(props) {
    super(props);
    this.state = { token: "null", loaded: true, loading: true,visible: true };

  }
  componentDidMount() {

  console.log("dfgdfgdfg0fgfgfgf000");
  this.props.getToken();
  if (this.props.categories) {
    Object.values(this.props.categories).map((category, index) => {
      this.props.loadCategoryProducts({
        token: this.props.token,
        departmentId: category.department_id,
        descriptionLength: 120,
        inStartItem: 0,
        inProductsPerPage: 10000
      });
      return category;
    });
  }
  }

  componentWillReceiveProps(props) {

    if (props.token) {
      if (this.state.token !== props.token) {
        this.setState({ token: props.token });
        if (!this.props.categories)
          this.props.getCategories({ token: props.token });
      }
    }
    if(props){
      this.setState({
        visible: false
      });
    }

  }
  render() {
    const { visible } = this.state;
    const { navigation } = this.props;
     const itemId = navigation.getParam('itemId', 'NO-ID');
     const categoryName = navigation.getParam('categoryName', 'NO-ID');
      console.log("Home screenffff",categoryName);
      return (
        <View>
          <NavBar />
        <ScrollView style={styles.body}>
        <View >
        <AnimatedLoader
        visible={visible}
        overlayColor="rgba(255,255,255,0.75)"
        source={require("../../lottie-loader.json")}
        animationStyle={{width: 100, height: 100}}
        speed={2}
      />
        <Text style={{...styles.h2, ...styles.black,...styles.cart_header}}>
          {categoryName}
        </Text>
        
        {this.props.categoryProducts ? Object.entries(this.props.categoryProducts).map(([key,productsList])=>{
          console.log(key,productsList,categoryName.toLowerCase()== key);
          if(categoryName.toLowerCase()== key){
                console.log(productsList);

                return  (<ProductList key={key} products={productsList} />);

          }
        }) : <Text/>}

        </View>
          <Footer />
        </ScrollView>
        </View>
        );
      }
}

const productRequest = Actions.products.request;
const checkUserLogin = Actions.checkUserLogin.request;

const mapStateToProps = state => ({
  subCategories: state.get("products").subCategories,
  categories: state.get("products").categories,
  categoryProducts: state.get("products").categoryProducts,
  cart: state.get("products").cart,
  token: state.get("user").token
});

const mapDispatchToProps = dispatch => ({
  productRequest,
  checkUserLogin,
  getCategories: data => dispatch(Actions.getCategories.request(data)),
  getCartProducts: inCartId =>
    dispatch(Actions.getCartProducts.request(inCartId)),
  getToken: () => dispatch(Actions.getToken.request()),
  loadCategoryProducts: data =>
    dispatch(Actions.getCategoryProducts.request(data))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Categories);
