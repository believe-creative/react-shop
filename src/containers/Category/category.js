import React, { Component } from "react";

export default class Category extends Component {
  componentDidMount() {}
  render() {
    return <div>Category:{this.props.match.params.category}</div>;
  }
}
