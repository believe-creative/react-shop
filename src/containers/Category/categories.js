import React, { Component } from "react";
import * as Actions from "../../actions";
import { connect } from "react-redux";
import ProductList from "../../components/Product/productlist";
import "../../scss/categories.scss";

class Categories extends Component {
  componentDidMount() {
    if (this.props.categories) {
      Object.values(this.props.categories).map((category, index) => {
        this.props.loadCategoryProducts({
          departmentId: category.department_id,
          descriptionLength: 120
        });
        return category;
      });
    }
  }

  render() {
    let productsList = [];
    let productCategoriesList = this.props.categoryProducts
      ? Object.values(this.props.categoryProducts)
      : [];
    productCategoriesList.map((category, index) => {
      category.map((item, ind) => {
        productsList.push(item);
        return item;
      });
      return category;
    });
    return (
      <div className="container">
        <div className="product_filter_panel">
          <div className="row">
            <div className="col-md-12 items_block">
              <div className="pb-3">
                <h2>ALL PRODUCTS</h2>
              </div>
              <section>
                {<ProductList products={productsList ? productsList : []} />}
              </section>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    subCategories: state.get("products").subCategories,
    categories: state.get("products").categories,
    categoryProducts: state.get("products").categoryProducts,
    searchitem: state.get("products").searchItem
  };
};

const mapStateToDispatch = dispatch => ({
  loadSubCategories: categoryId =>
    dispatch(Actions.getSubCategories.request(categoryId)),
  loadCategoryProducts: data =>
    dispatch(Actions.getCategoryProducts.request(data))
});

export default connect(
  mapStateToProps,
  mapStateToDispatch
)(Categories);
