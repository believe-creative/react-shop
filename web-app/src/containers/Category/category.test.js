import React from 'react';
import { shallow } from 'enzyme';

import Category from './category';
describe("Single Category", () => {
  it("should render all Category", () => {
    const wrapper = shallow(<Category />);
    expect(wrapper).toMatchSnapshot();
  });
});
