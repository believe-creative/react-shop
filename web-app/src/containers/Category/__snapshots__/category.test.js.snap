// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Category should render all Category 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Context.Consumer>
    [Function]
  </Context.Consumer>,
  "nodes": Array [
    <Context.Consumer>
      [Function]
    </Context.Consumer>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_didScheduleRenderPhaseUpdate": false,
    "_dispatcher": Object {
      "readContext": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useEffect": [Function],
      "useImperativeHandle": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useState": [Function],
    },
    "_element": <Connect(Category) />,
    "_firstWorkInProgressHook": null,
    "_forcedUpdate": false,
    "_instance": Connect {
      "context": Object {},
      "indirectRenderWrappedComponent": [Function],
      "props": Object {},
      "refs": Object {},
      "selectChildElement": [Function],
      "selectDerivedProps": [Function],
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_isReRender": false,
    "_newState": null,
    "_numberOfReRenders": 0,
    "_renderPhaseUpdates": null,
    "_rendered": <Context.Consumer>
      [Function]
    </Context.Consumer>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "_workInProgressHook": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Connect(Category) />,
}
`;
